examples/passing/ContextSimplification.purs

```diff
@@ -0,0 +1,13 @@
+module Main where
+
+import Prelude
+import Control.Monad.Eff.Console
+
+shout = log <<< (<> "!") <<< show
+
+-- Here, we should simplify the context so that only one Show
+-- constraint is added.
```

Comment:
Does this example test that the context is simplified - would it cause a type error if two `Show` constraints were added?